File: backend/test_api.py
import requests
import json

BASE_URL = "http://localhost:8000"

def test_api():
    print("=== Testing DrivoAI API ===\n")
    
    # Test 1: Voices endpoint
    print("[1] Testing /api/voices...")
    try:
        resp = requests.get(f"{BASE_URL}/api/voices")
        if resp.status_code == 200:
            voices = resp.json()
            print(f"    SUCCESS: {len(voices)} voices returned")
            for v in voices[:3]:
                print(f"      - {v['key']}: {v['name']} ({v['tone']})")
            print(f"      ... and {len(voices)-3} more")
        else:
            print(f"    FAILED: {resp.status_code}")
    except Exception as e:
        print(f"    ERROR: {e}")
    
    # Test 2: Login endpoint
    print("\n[2] Testing /api/auth/login...")
    try:
        resp = requests.post(f"{BASE_URL}/api/auth/login", json={
            "email": "xaviertheofilus@gmail.com",
            "password": "DrivoAI2024!"
        })
        if resp.status_code == 200:
            data = resp.json()
            print(f"    SUCCESS: Login works!")
            print(f"    User: {data['user']['email']}")
            token = data['access_token']
            print(f"    Token: {token[:30]}...")
        else:
            print(f"    FAILED: {resp.status_code} - {resp.text}")
    except Exception as e:
        print(f"    ERROR: {e}")
    
    # Test 3: Protected endpoint (analytics)
    print("\n[3] Testing /api/analytics (protected)...")
    try:
        resp = requests.get(f"{BASE_URL}/api/analytics")
        if resp.status_code == 401:
            print(f"    EXPECTED: Needs auth (401)")
        else:
            print(f"    Response: {resp.status_code}")
    except Exception as e:
        print(f"    ERROR: {e}")

if __name__ == "__main__":
    test_api()

File: frontend/test-api.js
const axios = require('axios');

const BACKEND_URL = process.env.REACT_APP_BACKEND_URL || 'http://localhost:8000';
const API = BACKEND_URL + '/api';

const api = axios.create({ baseURL: API, headers: { 'Content-Type': 'application/json' } });

const localStorage = {
  data: {},
  getItem(k) { return this.data[k] || null; },
  setItem(k, v) { this.data[k] = v; },
  removeItem(k) { delete this.data[k]; }
};

api.interceptors.request.use((config) => {
  const token = localStorage.getItem('drivoai_token');
  if (token) config.headers.Authorization = 'Bearer ' + token;
  return config;
});

async function test() {
  console.log('=== FRONTEND API TEST ===');
  console.log('Backend URL:', API);

  const email = 'new' + Date.now() + '@test.com';
  console.log('Email:', email);

  try {
    const r = await api.post('/auth/register', { email, password: 'Test1234!', full_name: 'Test User' });
    console.log('Step 1 - REGISTER: OK, token:', r.data.access_token ? 'YES (' + r.data.access_token.length + ' chars)' : 'NO');
    console.log('  User in response:', r.data.user?.email || 'MISSING');

    localStorage.setItem('drivoai_token', r.data.access_token);
    localStorage.setItem('drivoai_user', JSON.stringify(r.data.user));
    console.log('  Token saved to localStorage:', localStorage.getItem('drivoai_token') ? 'YES' : 'NO');

    const l = await api.post('/auth/login', { email, password: 'Test1234!' });
    console.log('Step 2 - LOGIN: OK, user:', l.data.user?.email || 'no user');

    const m = await api.get('/auth/me');
    console.log('Step 3 - /auth/me: OK, user:', m.data.email);

    const v = await api.get('/voices');
    console.log('Step 4 - VOICES: OK, count:', Array.isArray(v.data) ? v.data.length : 0);

    const s = await api.post('/sessions', { language: 'en' });
    console.log('Step 5 - SESSION: OK, session_id:', s.data.session_id || 'MISSING');
    console.log('  Voice config:', s.data.voice?.voice || 'MISSING');

    console.log('=== ALL TESTS PASSED ===');
  } catch(err) {
    console.log('ERROR at step:', err.response?.status, err.response?.data?.detail || err.message);
  }
}

test();